### PR TITLE
You can now store mice in your chef's hat.

### PR DESCRIPTION
**Pull request by ArcaneMusic**
## About The Pull Request

I sat there, playing chef, thinking about how I'll never be hired in a big time chef joint establishment, for several reasons (No professional interest in cooking, lack of skill, lack of professional cooking education, etc.) and then it hit me.
Why not train small, rodent creatures to help me cook better by storing them in my chef's hat?

## Why It's Good For The Game

To inspire a generation of future chefs into believing in themselves, to learn to better themselves, and to bring honor to the great french cooking community.

## Changelog
:cl:
add: Mice can now be stored in chef's hats.
/:cl:

### CODE REVIEW

**Comment by optimumtact** [comment]
this is an abstract shitpost

File: code/datums/components/storage/concrete/pockets.dm [hunk 1]
```diff
@@ -33,6 +33,25 @@
...
+	. = ..()
+	set_holdable(list(
+		/obj/item/clothing/head/mob_holder
+	))
+
+/datum/component/storage/concrete/pockets/chefhat/can_be_inserted(obj/item/I, stop_messages, mob/M)
+	. = ..()
+	if(istype(I,/obj/item/clothing/head/mob_holder))
+		var/obj/item/clothing/head/mob_holder/mausholder = I
+		for(var/mob/living/simple_animal/mouse/maus in mausholder.contents)
```
Comment (optimumtact) [inline code/datums/components/storage/concrete/pockets.dm:51]: just do a locate

**Comment by ATH1909** [comment]
Would it be possible to do this the "storing snakes and xenomorph larvae in boots" way too?

**Comment by ATH1909** [comment]
glorious

let chefs put dead mice in their hats as well, in case they need a sssssnack while away from their kitchen

**Comment by ArcaneMusic** [comment]
> glorious
> 
> let chefs put dead mice in their hats as well, in case they need a sssssnack while away from their kitchen

What no way that'd be unsanitary

**Comment by Winterous** [comment]
Please. Need a way to keep a backup mouse for mouse-replication shenanigans.

**Comment by OverlyGeneric** [comment]
Good PR.

Someone will eventually add the ability for mice to control the chef while inside the hat. God help us if they do.

**Comment by Caldony** [comment]
> 
> 
> Good PR.
> 
> Someone will eventually add the ability for mice to control the chef while inside the hat. God help us if they do.

step 1 : put sentient mice in chef hat
step 2 : throw the hat at your opponent 
step 3 : force your opponent to work in your illicit cooking practices

**Comment by zombiegums** [comment]
This made me smile.

File: code/datums/components/storage/concrete/pockets.dm [hunk 2]
```diff
@@ -33,6 +33,25 @@
...
+	))
+
+/datum/component/storage/concrete/pockets/chefhat/can_be_inserted(obj/item/I, stop_messages, mob/M)
+	. = ..()
+	if(istype(I,/obj/item/clothing/head/mob_holder))
+		var/obj/item/clothing/head/mob_holder/mausholder = I
+		if(locate(/mob/living/simple_animal/mouse) in mausholder.contents)
+			return TRUE
+		return FALSE
+
```
Comment (ShizCalev) [inline code/datums/components/storage/concrete/pockets.dm:54]: so real talk

```
/obj/item/clothing/head/chefhat/relaymove(mob/user, direction)
	if(istype(user, /mob/living/simple_animal/mouse)
		if(isliving(loc) && prob(20))
			var/mob/living/L = loc
			step_towards(L, direction)
```

**Review by ShizCalev — APPROVED** [review]

**Pull request merged**